docs/Definitions of Done/increase_event_visibility.md

```diff
@@ -0,0 +1,26 @@
+# Improve Event Visibility
+
+## Goals
+Make it easy to monitor events across multiple instances of Kōjō deployments
+1. Provide automatic logging of events as job's `assgined_state` is changed
+1. Provide consistent context about `kojo_job`, and `process` details in log events emitted from user space
+1. Provide visibility when any execution environment resource limit is being reached which is preventing more processes to be created
+ 
+## High Level Requirements
+* All jobs should emit an `info` log message when the `assigned_state` is modified in the `kojo_job` table.
```

Comment:
@alexberryman @mucha55 -

The only thing to keep in mind here is that y'all will need to make a decision on whether or not this event is emitted before or after the state change since it cannot be transactionally atomic.

Just a kindly reminder since we'll want the result of the decision to be explicit to our users.

fwiw  - I favor *after* since it is possible that a message can be lost but there is never an incorrect event.

Reply:
That's a really good point, job transitions within DB transactions is a pretty common occurrence. I could go either way, but, to play devil's advocate, I think there's value in not dropping messages and putting the responsibility on userspace to emit messages when they need to roll back a job transition transaction